tests/unit/models/governing-body-test.js

```diff
@@ -0,0 +1,828 @@
+      });
+
+      const changedStub = sinon.stub(model, 'changedAttributes');
+      changedStub.returns({});
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      changedStub.restore();
+    });
+
+    test('it should not check end date overlap unless it was changed', async function (assert) {
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-01-01'),
+        endDate: new Date('2024-02-02'),
+      });
+
+      const changedStub = sinon.stub(model, 'changedAttributes');
+      changedStub.returns({});
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      changedStub.restore();
+    });
+
+    test('it returns no overlap error when there are no other governing bodies', async function (assert) {
+      const unit = this.store().createRecord('administrative-unit', {
+        id: '1234',
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-02-01'),
+        endDate: new Date('2026-01-01'),
+        administrativeUnit: unit,
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
+  });
+
+  module('getOtherTimedGoverningBodies', function () {
+    test('it returns an empty array when there are no other timed governing bodies', async function (assert) {
+      const administrativeUnit = this.store().createRecord(
+        'administrative-unit',
+        { id: '1234' }
+      );
+
+      const classificationCode = this.store().createRecord(
+        'governing-body-classification-code',
+        { id: '5ab0e9b8a3b2ca7c5e000005' }
+      );
+      const untimedBody = this.store().createRecord('governing-body', {
+        classification: classificationCode,
+        administrativeUnit: administrativeUnit,
+      });
+
+      const timedBody = this.store().createRecord('governing-body', {
+        isTimeSpecializationOf: untimedBody,
+      });
+      untimedBody.hasTimeSpecializations.pushObject(timedBody);
+
+      const queryStub = sinon.stub(this.owner.lookup('service:store'), 'query');
+      queryStub.resolves(A([untimedBody]));
+
+      const governingBodies = await timedBody.getOtherTimedGoverningBodies();
+      assert.strictEqual(governingBodies.length, 0);
+
+      queryStub.restore();
+    });
+
+    test('it returns an array containing the other timed governing bodies when there is one of the same classification', async function (assert) {
+      const administrativeUnit = this.store().createRecord(
+        'administrative-unit',
+        { id: '1234' }
```

Comment:
**note:** I notice that you are adding `id` in almost all `createRecord` but they are not required, unless you need to use them in the test they can be removed.

Reply:
The set `id` is used in `#getAdministrativeUnitId` which is called in `#getUntimedGoverningBodies` which  in turn is called by the function being tested `getOtherGoverningBodies`. Not specifying an `id` causes this functionality, and thereby also the tests, to fail in an unintended way.